src/panels/lovelace/editor/unused-entities/hui-unused-entities.ts

```diff
@@ -86,54 +75,58 @@ export class HuiUnusedEntities extends LitElement {
             : ""}
         </div>
       </ha-card>
-      <search-input @value-changed=${this._handleSearchChange}></search-input>
       <ha-data-table
-        .headers=${[
-          {
-            name: "Entity",
+        .columns=${{
+          entity: {
+            title: "Entity",
             sortable: true,
-            direction: this._sortColumn === 0 ? this._sortDirection : null,
+            filterable: true,
+            filterKey: "friendly_name",
+            direction: "desc",
+            template: (stateObj) => html`
+              <state-badge
+                .hass=${this.hass!}
+                .stateObj=${stateObj}
+              ></state-badge>
+              ${stateObj.friendly_name}
+            `,
           },
-          {
-            name: "Entity id",
+          id: {
+            title: "Entity id",
             sortable: true,
-            direction: this._sortColumn === 1 ? this._sortDirection : null,
+            filterable: true,
           },
-          {
-            name: "Domain",
+          domain: {
+            title: "Domain",
             sortable: true,
-            direction: this._sortColumn === 2 ? this._sortDirection : null,
+            filterable: true,
           },
-          {
-            name: "Last Changed",
+          last_changed: {
+            title: "Last Changed",
             type: "numeric",
             sortable: true,
-            direction: this._sortColumn === 3 ? this._sortDirection : null,
-          },
-        ]}
-        .data=${this._sortedUnusedEntities.map((entity) => {
-          const stateObj = entity ? this.hass!.states[entity] : undefined;
-          return [
-            html`
-              <state-badge
-                .hass=${this.hass!}
-                .stateObj=${stateObj}
-              ></state-badge>
-              ${computeStateName(stateObj!)}
-            `,
-            entity,
-            computeDomain(entity),
-            html`
+            template: (lastChanged: string) => html`
               <ha-relative-time
                 .hass=${this.hass!}
-                .datetime=${stateObj!.last_changed}
+                .datetime=${lastChanged}
               ></ha-relative-time>
             `,
-          ];
+          },
+        }}
+        .data=${this._unusedEntities.map((entity) => {
```

Comment:
This is expense. However, we don't need to memoize this because right now either `lovelace` or `hass` changes, we want to completely rebuild .